wasm/src/main/java/com/dylibso/chicory/wasm/Module.java

```diff
@@ -151,4 +166,150 @@ public ElementSection elementSection() {
     public void setElementSection(ElementSection elementSection) {
         this.elementSection = requireNonNull(elementSection);
     }
+
+    /**
+     * Creates a {@link Builder} for the specified {@link InputStream}
+     *
+     * @param input the input stream
+     * @return a {@link Builder} for reading the module definition from the specified input stream
+     */
+    public static Builder builder(InputStream input) {
+        return new Builder(() -> input);
+    }
+
+    /**
+     * Creates a {@link Builder} for the specified {@link com.dylibso.chicory.wasm.Module}
+     *
+     * @param wasmModule the already parsed Wasm module
+     * @return a {@link Builder} for reading the module definition from the specified input stream
+     */
+    public static Builder builder(com.dylibso.chicory.wasm.Module wasmModule) {
+        return new Builder(wasmModule);
+    }
+
+    /**
+     * Creates a {@link Builder} for the specified {@link ByteBuffer}
+     *
+     * @param buffer the buffer
+     * @return a {@link Builder} for reading the module definition from the specified buffer
+     */
+    public static Builder builder(ByteBuffer buffer) {
+        return builder(buffer.array());
+    }
+
+    /**
+     * Creates a {@link Builder} for the specified byte array
+     *
+     * @param buffer the buffer
+     * @return a {@link Builder} for reading the module definition from the specified buffer
+     */
+    public static Builder builder(byte[] buffer) {
+        return new Builder(() -> new ByteArrayInputStream(buffer));
+    }
+
+    /**
+     * Creates a {@link Builder} for the specified {@link File} resource
+     *
+     * @param file the path of the resource
+     * @return a {@link Builder} for reading the module definition from the specified file
+     */
+    public static Builder builder(File file) {
+        return new Builder(
+                () -> {
+                    try {
+                        return new FileInputStream(file);
+                    } catch (FileNotFoundException e) {
+                        throw new IllegalArgumentException(
+                                "File not found at path: " + file.getPath(), e);
+                    }
+                });
+    }
+
+    /**
+     * Creates a {@link Builder} for the specified classpath resource
+     *
+     * @param classpathResource the name of the resource
+     * @return a {@link Builder}  for reading the module definition from the specified resource
+     */
+    public static Builder builder(String classpathResource) {
+        return new Builder(
+                () -> {
+                    InputStream is =
+                            Thread.currentThread()
+                                    .getContextClassLoader()
```

Comment:
Relying on the thread context class loader is problematic, as we saw with `Wast2Json`. We should consider removing this API and let callers use `Class.getResourceAsStream()` or similar, as appropriate. This avoids having a hidden dependency on the context class loader. If the user wants that, it will be explicit in their application code.

Reply:
I agree, doing the changes